Improve a11y on the cart and sidebar

Fedeorlandau opened the pull request:
## Improving a11y on the cart and sidebar

This PR includes:
* A new Button variant `Naked` to wrap SVG buttons. Previously the we had the SVG only.
* Now we can focus on the cart button
* We can close the sidebar by pressing `Escape`
* Focus the sidebar on open.

![nextjs](https://user-images.githubusercontent.com/10283686/124911521-20795d80-dfed-11eb-8460-3202d0a3242b.gif)


Regards,

Fedeorlandau commented on components/ui/Sidebar/Sidebar.tsx:
```diff
@@ -13,27 +13,44 @@ interface SidebarProps {
   const ref = useRef() as React.MutableRefObject<HTMLDivElement>
 
+  const onKeyDownSidebar = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.code === 'Escape') {
```
Please let me know if you find a better way to check this.

lfades reviewed (changes requested):
The cart icon has a different background: https://commerce-bigcommerce-git-fork-8f3229-vercel-solutions-vtest314.vercel.app/

How it should be: https://commerce-bigcommerce-git-fork-438b79-vercel-solutions-vtest314.vercel.app/

Fedeorlandau commented:
> Hi there and thank you for the PR!
> 
> The cart icon has a different background: https://commerce-bigcommerce-git-fork-8f3229-vercel-solutions-vtest314.vercel.app/
> 
> How it should be: https://commerce-bigcommerce-git-fork-438b79-vercel-solutions-vtest314.vercel.app/

Fixed. I've just added the `s.item` class to the button to get the dark/light mode behavior.

![image](https://user-images.githubusercontent.com/10283686/127648995-4a623b42-14a5-4b81-ac21-fd5028a54103.png)

![image](https://user-images.githubusercontent.com/10283686/127649039-4d89f642-cc80-47e2-b42f-def5a1dd3a96.png)

Thanks for your feedback

Fedeorlandau commented:
Hi @lfades I've fixed the conflicts the the main branch. Hope this get merged soon :) Thanks

lfades commented:
@Fedeorlandau LGTM. Thank you and amazing work!
